app/models/annotation.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Annotation < ApplicationRecord
+  SUPPORTED_ANNOTATIONS = %w[managed].freeze
+
+  belongs_to :annotated, polymorphic: true
+
+  validates :name, presence: true, inclusion: { in: SUPPORTED_ANNOTATIONS }
+  validates :name, :value, :annotated_type, length: { maximum: 255 }
```

Comment:
Maybe annotated_type no need to be checked for length if we have `required: true` above.

Reply:
I had to add the length validation because a [test](https://app.circleci.com/pipelines/github/3scale/porta/29066/workflows/d378dac6-f0a7-46ab-8b79-762462e8d68e/jobs/325483/tests) was failing.